### ISSUE TITLE
Crash while using Combined Chart

### ISSUE THREAD

**Issue by vaibhav-varshaaweblabs** [danielgindi/Charts#1540]
Charts crashes while populating the dataset 

`let set: BarChartDataSet = BarChartDataSet.init(yVals: userTimesData, label: "User Times")`

Whole function - 

```
func populateBarChartData(userTimes: [Double?]) -> BarChartData {
        let barChartData: BarChartData = BarChartData()
        
        var userTimesData : [ChartDataEntry] = [ChartDataEntry]()
        
        for i in 0 ..< userTimes.count {
            userTimesData.append(ChartDataEntry(value: userTimes[i]!, xIndex: i))
        }
        
        let set: BarChartDataSet = BarChartDataSet.init(yVals: userTimesData, label: "User Times")
        set.axisDependency = .Left
        set.setColor(UIColor.greenColor().colorWithAlphaComponent(0.5))
        set.highlightColor = UIColor.whiteColor()
        
        var dataSets : [BarChartDataSet] = [BarChartDataSet]()
        dataSets.append(set)
        
        barChartData.dataSets = dataSets
        
        return barChartData
    }
```

**Comment by liuxuan30** [comment]
you say it crashes but why don't you paste the crash log?

**Comment by petester42** [comment]
This seems like it could be the same issue as #1550 which has been addressed by #1558. Is userTimes an empty array when it crashes?
